ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -125,8 +130,8 @@ def attach(self, engine, metric_names=None, output_transform=None):
                 output. This function may return either a dictionary with entries in the format of ``{name: value}``,
                 or a single scalar, which will be displayed with the default name `output`.
         """
-        if metric_names is not None and not isinstance(metric_names, list):
-            raise TypeError("metric_names should be a list, got {} instead".format(type(metric_names)))
+        if metric_names is not None and not isinstance(metric_names, _collections_abc.Sequence):
```

Comment:
This will also allow incorrect types like string to be passed through (which is why the build is failing as we have a test for this)

Reply:
I've modified the guard statement to account for strings. Would this be better with a `try`/`except` block though? That is, assume that the client gave us something valid for `metric_names` and catch any exceptions that are raised and raise our own `TypeError`?